Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -552,8 +589,17 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
   for (auto i = 0u; i < sizeOfLabels; ++i) {
     if (m_header.labelsOfHistograms[i] == "")
       break;
-    addToSampleLog("Label Spectra " + std::to_string(i),
-                   m_header.labelsOfHistograms[i], outputWorkspace);
+    std::string name = m_header.labelsOfHistograms[i];
+    // if no name is present (or just empty space)
+    // replace with default name:
+    // group_specNum
+    std::string label =
+        (name.size() > 0 && name.find_first_not_of(" ") == std::string::npos)
```

Comment:
Nitpick: I don't like this line it's doing too much and can be simplified a touch (since the STL does the name size check for us)
```
bool is_empty = name.find_first_not_of(" ") == std::string::npos;
std::string label = is_empty ? "group_..." : m_header.labelsOfHistrograms[i]; 
```

Reply:
STL?